src/utils/GitLabUtils/GitLabUtils.test.js

```diff
@@ -0,0 +1,68 @@
+import { expect } from 'chai';
+import * as GitLabUtils from './GitLabUtils';
+import * as TestUtils from '../TestUtils/TestUtils';
+
+describe('GitLabUtils', () => {
+  describe('mergeCalendarsContributions', () => {
+    const actualCalendarFirstDate = '2019-04-18';
+    const contributionsOnFirstDate = 5;
+
+    const actualCalendar = TestUtils.getFakeContributionsObjectWithDailyCounts(
+      [contributionsOnFirstDate],
+      [actualCalendarFirstDate],
+    )[0];
+
+    describe('when the user`s JSON calendar contains contributions on `actualCalendarFirstDate`', () => {
+      const calendarWithContributionsOnFirstDate = {
+        '2019-04-18': 7,
+      };
+
+      it('increments the first date`s `data-count` property by the user`s contributions on `actualCalendarFirstDate`', () => {
+        const expectedDataCountValue = String(
+          contributionsOnFirstDate + calendarWithContributionsOnFirstDate[actualCalendarFirstDate],
+        );
+
+        const updatedActualCalendar = GitLabUtils.mergeCalendarsContributions(
+          actualCalendar, calendarWithContributionsOnFirstDate,
+        );
+
+        const actualDataCountValue = updatedActualCalendar.children[0].children[0].children[0].attributes['data-count'];
+
+        expect(actualDataCountValue).to.equal(expectedDataCountValue);
+      });
+    });
+
+    describe('when the user`s JSON calendar does not contain contributions on `actualCalendarFirstDate`', () => {
+      const calendarWithoutContributionsOnFirstDate = {
```

Comment:
Whenever a user doesn't own contributions on a particular date then the GL calendar doesn't store the related date-contributions pair in the object.

```
// E.g. this data isn't available in the structure.
{
   '2019-02-20': 0,
}
```

That's the reason for creating a test calendar with a different date - instead of just setting its value to 0.